### PR TITLE
[WIP] scalar support for unary ndarray, ie. nd.sqrt(2), nd.log(3)

### PR DESCRIPTION
**Pull request by ZihengJiang**
ref https://github.com/dmlc/mxnet/issues/4844 

BTW, it looks like MXNet.NDArray only accept array-like data, so `nd.sqrt(2)` will return `array([1.41421354])` instead of `array(1.41421354)`

### CODE REVIEW

**Comment by piiswrong** [comment]
why not just float(1.414)

File: python/mxnet/_ctypes/ndarray.py [hunk 1]
```diff
@@ -97,6 +121,11 @@ def generic_ndarray_function(*args, **kwargs):
         for i in args:
             if isinstance(i, NDArrayBase):
                 ndargs.append(i)
+            elif num_args.value == 1:
```
Comment (piiswrong) [inline python/mxnet/_ctypes/ndarray.py]: This isn't valid. It breaks mx.nd.arange()
I think you should fix this by falling back on minpy side

Comment (piiswrong) [inline python/mxnet/_ctypes/ndarray.py]: or at least check if this argument type is NDArray|Symbol

**Comment by ZihengJiang** [comment]
yes, I think it's easy to solve on the MinPy side, but do you agree to make MXNet.NDArray to support such usage? @piiswrong

**Comment by piiswrong** [comment]
We can make NDArray support scalar, but you need to check for argument type ("NDArray or Symbol") to make sure it's not meant to be a kwarg

**Comment by piiswrong** [comment]
Can one of the admins verify this patch?

**Comment by mli** [comment]
closed due to no recent activity. please feel free to reopen it.

**Pull request closed without merging**